analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.Operations.cs

```diff
@@ -125,19 +126,40 @@ public void SimpleAssignment_Fields(string snippet, string operation)
         }
 
         [DataTestMethod]
-        [DataRow(@"Sample.StaticProperty = 42; Tag(""Target"", Sample.StaticProperty);")]
-        [DataRow(@"StaticProperty = 42; Tag(""Target"", StaticProperty);")]
-        [DataRow(@"var arr = new byte[] { 13 }; arr[0] = 42; Tag(""Target"", arr[0]);")]
+        [DataRow(@"this.AutoProperty = 42; Tag(""Target"", this.AutoProperty);")]
+        [DataRow(@"AutoProperty = 42; Tag(""Target"", AutoProperty);")]
+        [DataRow(@"Sample.StaticAutoProperty = 42; Tag(""Target"", Sample.StaticAutoProperty);")]
+        [DataRow(@"StaticAutoProperty = 42; Tag(""Target"", StaticAutoProperty);")]
+        public void SimpleAssignment_FromLiteral(string snippet)
+        {
+            var validator = SETestContext.CreateCS(snippet, new LiteralDummyTestCheck()).Validator;
+            validator.Validate("Literal: 42", x => x.State[x.Operation].Should().HaveOnlyConstraints(new SymbolicConstraint[] { ObjectConstraint.NotNull, DummyConstraint.Dummy }, "it's scaffolded"));
+            validator.ValidateTag("Target", x => x.Should().HaveOnlyConstraints(ObjectConstraint.NotNull, DummyConstraint.Dummy));
+        }
+
+        [DataTestMethod]
+        [DataRow(@"Sample.StaticFullProperty = 42; Tag(""Target"", Sample.StaticFullProperty);")]
+        [DataRow(@"StaticFullProperty = 42; Tag(""Target"", StaticFullProperty);")]
         [DataRow(@"var dict = new Dictionary<string, int>(); dict[""key""] = 42; Tag(""Target"", dict[""key""]);")]
-        [DataRow(@"var other = new Sample(); other.Property = 42; Tag(""Target"", other.Property);")]
-        [DataRow(@"this.Property = 42; Tag(""Target"", this.Property);")]
-        [DataRow(@"Property = 42; Tag(""Target"", Property);")]
+        [DataRow(@"var other = new Sample(); other.AutoProperty = 42; Tag(""Target"", other.AutoProperty);")]
+        [DataRow(@"var other = new Sample(); other.FullProperty = 42; Tag(""Target"", other.FullProperty);")]
+        [DataRow(@"this.FullProperty = 42; Tag(""Target"", this.FullProperty);")]
+        [DataRow(@"FullProperty = 42; Tag(""Target"", FullProperty);")]
         [DataRow(@"var other = new Sample(); other.field = 42; Tag(""Target"", other.field);")]
         public void SimpleAssignment_ToUnsupported_FromLiteral(string snippet)
```

Comment:
I guess this is called `ToUnsupported`, and not `Unsupported` because what is not supported is not the assignment itself, but the assignment to a target such a property of another instance, where "target such a property of another instance" is implicit, to keep the text shorter. 
That took me a while to grasp, from the name of the method, and maybe `SimpleAssignment_Unsupported` would be both more grammar-conforming and easier to understand.

Reply:
What about `SimpleAssignment_FromLiteral_ToUntracked`?